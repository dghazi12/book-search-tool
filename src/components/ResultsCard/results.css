.results-container {
  display: flex;
  margin-left: auto;
  margin-right: auto;
  width: 40%;
  margin-top: 20px;
}

.image-style {
  height: 300px;
  width: 200px;
}

.book-title {
  font-size: 20px;
  font-style: normal;
  margin-left: 10px;
}

.book-info {
  font-size: 18px;
  font-style: normal;
  font-weight: 400;
  margin: 10px 20px 0 10px;
  width: 90%;
}

@media screen and (max-width: 768px) {
  .image-style {
    height: 250px;
    width: 150px;
  }

  .book-title {
    font-size: 16px;
  }

  .book-info {
    font-size: 14px;
  }
}

@media screen and (max-width: 600px) {
  .image-style {
    height: 180px;
    width: 100px;
  }

  .book-title {
    font-size: 14px;
  }

  .book-info {
    font-size: 10px;
  }
}

@media screen and (max-width: 375px) {
  .results-container {
    width: 50%;
  }
}
