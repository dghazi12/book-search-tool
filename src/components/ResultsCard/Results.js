import "./results.css";
import NotFound from "../../images/no-image-found.jpeg";

const Results = ({ image, title, author, publication }) => {
  return (
    <div className="results-container">
      <div className="img-container">
        <img
          alt={`${title} cover`}
          className="image-style"
          src={
            image
              ? `http://covers.openlibrary.org/b/id/${image}-M.jpg`
              : NotFound
          }
        />
      </div>

      <div className="info-container">
        <h2 className="book-title">{title} </h2>
        <p className="book-info">Author: {author}</p>
        <p className="book-info">Publication: {publication} </p>
      </div>
    </div>
  );
};

export default Results;
