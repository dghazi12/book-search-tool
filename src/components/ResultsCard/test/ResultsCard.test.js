import React from "react";
import ResultsCard from "../Results";
import { render } from "@testing-library/react";

it("renders correctly", () => {
  render(<ResultsCard />);
});

const dataExample = {
  image: 9269962,
  title: "A Game of Thrones",
  author: "George R. R. Martin",
  PubDate: 1996,
};

it("matches snapshot", () => {
  const { asFragment } = render(
    <ResultsCard
      image={dataExample.image}
      title={dataExample.title}
      author={dataExample.author}
      PubDate={dataExample.PubDate}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
