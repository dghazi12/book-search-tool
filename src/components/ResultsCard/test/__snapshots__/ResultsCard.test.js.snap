// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="results-container"
  >
    <div
      class="img-container"
    >
      <img
        alt="A Game of Thrones cover"
        class="image-style"
        src="http://covers.openlibrary.org/b/id/9269962-M.jpg"
      />
    </div>
    <div
      class="info-container"
    >
      <h2
        class="book-title"
      >
        A Game of Thrones 
      </h2>
      <p
        class="book-info"
      >
        Author: George R. R. Martin
      </p>
      <p
        class="book-info"
      >
        Publication:  
      </p>
    </div>
  </div>
</DocumentFragment>
`;
